File: 백준/Silver/10828. 스택/스택.java
import java.util.Scanner;
import java.util.Stack;

public class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		StringBuffer sb = new StringBuffer();
		Stack<Integer> stack = new Stack<>();
		int n = sc.nextInt();
		
		for (int i = 0; i < n; i++) {
			String order = sc.next();
			
			switch (order) {
			case "push":
				stack.push(sc.nextInt());
				break;
			case "pop":
				if(stack.isEmpty()) 
					sb.append(-1).append('\n');
				else sb.append(stack.pop()).append('\n');
				break;
			case "size":
				sb.append(stack.size()).append('\n');
				break;
			case "empty":
				if(stack.isEmpty())
					sb.append(1).append('\n');
				else sb.append(0).append('\n');
				break;
			case "top":
				if (stack.isEmpty()) sb.append(-1).append('\n');
				else sb.append(stack.peek()).append('\n');
				break;
			}
		}
		
		System.out.println(sb);
	}
}

File: 백준/Silver/10828. 스택/스택.cc
#include <iostream>
#include <stack>
#include <string>
using namespace std;


int main()
{
	ios::sync_with_stdio(false);
	cin.tie(nullptr);

	int N;
	cin >> N;

	stack<int> stack;

	for (int i = 0; i < N; i++)
	{
		string str;
		cin >> str;

		if (str == "push")
		{
			int x;
			cin >> x;

			stack.push(x);
		}
		else if (str == "top")
		{
			if (stack.size() <= 0)
			{
				cout << -1 << "\n";
			}
			else
			{
				cout << stack.top() << "\n";
			}
		}
		else if (str == "pop")
		{
			if (stack.size() <= 0)
			{
				cout << -1 << "\n";
			}
			else
			{
				cout << stack.top() << "\n";
				stack.pop();
			}
		}
		else if (str == "size")
		{
			cout << stack.size() << "\n";
		}
		else
		{
			if (stack.empty())
			{
				cout << 1 << "\n";
			}
			else
			{
				cout << 0 << "\n";
			}
		}
	}
}